content/en/blog/2023/2023-03-08-scheduler-plugin-reusing/index.md

```diff
@@ -0,0 +1,124 @@
+---
+layout: blog
+title: "Scheduler plugin reusing"
+---
+
+**Author:** Ukri Niemimuukko (Intel)
+
+The Kubernetes Scheduler comes with a well-established [scheduling framework](https://kubernetes.io/docs/concepts/scheduling-eviction/scheduling-framework/) which allows for creating both in-tree and out-of-tree plugins. By the nature and design of the plugins, they do not have much in terms of dependencies between each other. This is intentional and a good thing, plugins can be disabled and enabled at will. But even such entities which aim to be cohesive and without coupling, could occasionally benefit from being able to reuse the code from existing plugins.
+
+## The use-case
+
+A particular use-case for such plugin reuse arose from the need of getting a score for topologies within a cluster based on the resource status of the topology, such as a rack, instead of scoring just the resource status of the individual nodes in that rack. The aim there, was to be able to bin-pack Pods to racks, and at other times to be able to find a rack for a Pod with very little resources in use at the moment. A plugin which gives the nodes in the best rack a score of 100, and the nodes in the worst rack a score of 0, and then gives nodes in other racks scores linearly between the two, would enable that.
+
+Resource calculations and scoring of nodes is currently done quite nicely and with a well configurable interface within the _NodeResourcesFit_ plugin. It can be [configured](https://kubernetes.io/docs/reference/config-api/kube-scheduler-config.v1/#kubescheduler-config-k8s-io-v1-NodeResourcesFitArgs) to use different scoring strategies such as _LeastAllocated_ and _MostAllocated_. It does the job, and it does it well. Reinventing the wheel which has been implemented in _NodeResourcesFit_ for calculating of node resources, would seem like a lot of unnecessary, duplicated work. Ideally, one would take up topology resource calculations and scoring from where _NodeResourcesFit_ plugin left off, and form the topology score based on the scores already calculated for individual nodes.
+
+## Alternative approaches for plugin reuse
+
+### Ctrl-C, Ctrl-V
+
+> **Warning**
+> The simplest and probably most frowned upon code-reuse practice is of course copy-paste. Due to well known issues with copy-paste code, this approach can't be recommended, and is mentioned here only as an example of what not to do. Copy-pasting won't be any faster anyway, as you will see below.
```

Comment:
maybe remote the `>` symbol? otherwise this section is entirely quoted without any regular paragraph, which looks a bit strange.

Reply:
Ok, will do.